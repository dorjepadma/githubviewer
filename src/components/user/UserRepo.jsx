import React from 'react';
import PropTypes from 'prop-types';
import User from './User.jsx';

const CompleteUser = ({ completeUser }) => {
  // const completeUserRepo = completeUser.map(user => (
  //   <li key={user.id}>
  //     <User {...user} />
  //   </li>
  // ));

  return (
    <ul>
      {completeUserRepo}
    </ul>
  );
};
CompleteUser.propTypes = {
  completeUser: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    followers: PropTypes.string.isRequired,
    following: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    repos_url: PropTypes.string.isRequired,
  })).isRequired
};
export default CompleteUser;
 
