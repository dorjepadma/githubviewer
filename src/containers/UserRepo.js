import React, { Component } from 'react';
import User from '../components/user/User.jsx';
import { fetchUser } from '../services/github.js';

export default class FinalUser extends Component {
  state = {
    user: [],
    loading: true,
  }
  componentDidMount() {
    console.log('test')
    fetchUser()
      .then(user => this.setState({ user, loading: false }
      ))
  }
  render() {
    const { user, loading } = this.state;
    if(loading) return <h1>Patience</h1>;

    return (
      <User {...user} />
    );
  }
}

