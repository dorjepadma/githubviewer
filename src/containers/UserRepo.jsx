import React, { Component } from 'react';
import Repos from '../components/Repos/Repos';
import { fetchUserRepos } from '../services/github.js';
import PropTypes from 'prop-types';

export default class UserRepos extends Component {
  static propTypes = {
    username: PropTypes.string
  }
  
  state = {
    repos: [],
    loading: true,
  }
  fetchUserRepos = () => {
    if(!this.props.username) return;

    this.setState({ loading: true });
    fetchUserRepos(this.props.username)
      .then(repos =>{
        this.setState({ repos, loading: false });
      });
  }
  componentDidMount() {
    this.fetchUserRepos();
  }
  componentDidUpdate(prevProps) {
    if(prevProps.username !== this.props.username) {
      this.fetchUserRepos();
    }
  }
  render() {
    const { repos } = this.state;

    return (
      <Repos repos={repos} />
    );
  }
}

