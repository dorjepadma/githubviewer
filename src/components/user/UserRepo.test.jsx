import React from 'react';
import { shallow } from 'enzyme';
import UserRepo from '../user/UserRepo.jsx';

describe('UserRepo', () => {
  it('matches a snapshot', () => {
    const repo = [
      { id:'1',
        name: 'dorjepadma',
        followers: '1',
        following: '1',
        url: 'https://api.github.com/users/dorjepadma',
        avatar_url: 'https://avatars0.githubusercontent.com/u/59029366?v=4',
        repos_url: 'https://api.github.com/users/dorjepadma/repos' }
    ];
    const wrapper = shallow(<UserRepo completeUser = {repo}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
