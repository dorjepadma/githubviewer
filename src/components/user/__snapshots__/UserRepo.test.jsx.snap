// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRepo matches a snapshot 1`] = `
<ul>
  <li
    key="1"
  >
    <User
      avatar_url="https://avatars0.githubusercontent.com/u/59029366?v=4"
      followers="1"
      following="1"
      id="1"
      name="dorjepadma"
      repos_url="https://api.github.com/users/dorjepadma/repos"
      url="https://api.github.com/users/dorjepadma"
    />
  </li>
</ul>
`;
